[WIP] LLVM Value Profiler

ksmonkey123 opened the pull request.

ksmonkey123 committed:
basic type recording in LLVM

ksmonkey123 committed:
basic type recording in LLVM

ksmonkey123 committed:
fixed malloc in signal handler

ksmonkey123 committed:
fixed malloc in signal handler

ksmonkey123 committed:
fixed apple build

ksmonkey123 committed:
fixed apple build

o- committed:
opt: increasing number of passes and stop when done

increse the number of passes but at the same time stop optimizing
when there are no more significant changes happening.

o- committed:
fix rir_check

o- committed:
drive-by fixes to the fast ranges

o- committed:
Merge pull request #859 from reactorlabs/optSchedule2

opt: increasing number of passes and stop when done

o- committed:
some more bugs...

o- committed:
make sure those tests don't take forever

o- committed:
Merge pull request #861 from reactorlabs/fixIntSeq2

drive-by fixes to fst ranges

ksmonkey123 committed:
configured for more lenient testing

ksmonkey123 committed:
configured for more lenient testing

ksmonkey123 committed:
re-added previous test

ksmonkey123 committed:
re-added previous test

ksmonkey123 committed:
Merge branch 'valueProfiler' of github.com:reactorlabs/rir into valueProfiler

o- commented:
@ksmonkey123 after merging travis should be green too. so if you want I can merge it.

o- committed:
pass sampled typefeedback into compiler

ksmonkey123 committed:
initial version of reopting

o- committed:
rename PirVariableMap to pirTypeFeedback

o- committed:
save the bytes

save a couple of bytes in the typefeedback by removing redundant
information from the entries.

also provide access to the code object of each entry

o- committed:
add a simple test for the value profiler

o- committed:
fix the signal handler

the signal handler is quite brittle. it seems like the previous version
did again do some allocation which corrupts malloc metadata. we probably
need a more robust solution here.
